source/Government.cpp

```diff
@@ -400,3 +427,38 @@ bool Government::IsProvokedOnScan() const
 {
 	return provokedOnScan;
 }
+
+
+
+bool Government::ShowReputation() const
+{
+	return showReputation;
+}
+
+
+
+void Government::SetShowReputation(bool input)
+{
+	showReputation = input;
+}
+
+
+
+string Government::GetReputationTitle() const
+{
+	double rep = Reputation();
+	auto tempTitles = titles;
+	if(tempTitles.empty())
+		tempTitles = *GameData::ReputationTitles();
+	if(tempTitles.empty())
+		return "";
+	string title;
+	for(auto item : tempTitles)
+	{
+		double lowerBound = item.first;
+		if(rep < lowerBound)
+			break;
+		title = item.second;
```

Comment:
Maybe this is the source of the issue I mentioned above?
Even if it isn't I don't think we should assume that the reputation is given in any specific order, so either we sort it or this should be changed.

Reply:
It is now using an `std::map` so the traversal order will be in ascending order.